canvas/image.go

```diff
@@ -100,7 +100,9 @@ func (i *Image) Hide() {
 // MinSize returns the specified minimum size, if set, or {1, 1} otherwise.
 func (i *Image) MinSize() fyne.Size {
 	if i.Image == nil || i.aspect == 0 {
-		i.Refresh()
+		if i.File != "" || i.Resource != nil {
```

Comment:
You can move this into the upper if-statement to remove some unnecessary nesting.

Reply:
I agree that it could - but it didn't feel right because they are checking different things. Outer is "do we have valid state" and inner is "can we do anything about it".

Reply:
Good point